File: FormalBench-data/FilteredData/specifications/seed_726_per_category_10_653ade686f/c/TestThreeEqual.c
#include "java_arrays.h"
#include <stdint.h>

/*@
  assigns \nothing;
  ensures \result == (x == y && y == z ? 3 : (x == y || y == z || x == z ? 2 : 0));
*/
int32_t testThreeEqual(int32_t x, int32_t y, int32_t z) {
    if (x == y && y == z) {
        return 3;
    } else if (x == y || y == z || x == z) {
        return 2;
    }
    return 0;
}

File: FormalBench-data/FilteredData/specifications/seed_726_per_category_10_653ade686f/java/TestThreeEqual.java
import java.io.*;
import java.lang.*;
import java.util.*;
import java.math.*;


class TestThreeEqual {
    
    /*@ public normal_behavior
      @ assignable \nothing;
      @ ensures \result == (x == y && y == z ? 3 : (x == y || y == z || x == z ? 2 : 0));
      @*/
    //@ code_java_math
    public static int testThreeEqual(int x, int y, int z) {
        if(x == y && y == z)
            return 3;
        else if(x == y || y == z || x == z)
            return 2;
        else
            return 0;
    }
}

